x-pack/plugins/apm/server/lib/helpers/create_es_client/create_apm_event_client/index.ts

```diff
@@ -149,11 +152,18 @@ export class APMEventClient {
       this.indices
     );
 
+    const forceSyntheticSourceForThisRequest =
+      this.forceSyntheticSource &&
+      params.apm.events.includes(ProcessorEvent.metric);
```

Comment:
If an index is not synthetic source-compatible, I assume it results in a shard failure, but only when actually fetching the documents. E.g., if `size` is 0 or the top scoring documents are retrieved from compatible indices, it still works. I prefer this approach over only using `force_synthetic_source` when only the metrics data stream is being queried, because this currently works, and I'd like to eliminate false positives over preventing false negatives. At least we get notified of the latter.